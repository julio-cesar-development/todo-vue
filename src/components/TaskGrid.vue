<template>
  <div class="task-grid">
    <template v-if="tasks && tasks.length">
      <task v-for="(task, idx) in tasks"
        v-bind:task="task"
        v-bind:key="task.name"
        v-on:task-deleted="deleteTask(idx)"
        v-on:task-state-changed="changeTaskState(idx)"
      />
    </template>
    <template v-else>
      <div>
        <p class="no-task">Nenhuma task ðŸ˜€</p>
      </div>
    </template>
  </div>
</template>

<script>
import Task from '@/components/Task.vue';

export default {
  name: 'TaskGrid',
  components: {
    Task,
  },
  data() {
    return {
      value: '',
    };
  },
  props: {
    tasks: {
      type: Array,
      required: true,
    },
  },
  methods: {
    deleteTask(index) {
      if (index !== undefined && index !== null && index !== '') {
        this.$emit('task-deleted', index);
      }
    },
    changeTaskState(index) {
      if (index !== undefined && index !== null && index !== '') {
        this.$emit('task-state-changed', index);
      }
    },
  },
};
</script>

<style scoped>
.task-grid {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  padding: 10px;
}

.task-grid .task {
  margin: 10px;
}

.no-task {
  color: #bbb;
  font-size: 1.8em;
}
</style>
