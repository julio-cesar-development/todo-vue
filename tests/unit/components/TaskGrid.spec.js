import { shallowMount } from '@vue/test-utils';
import TaskGrid from '@/components/TaskGrid.vue';

describe('TaskGrid.vue', () => {
  it('TaskGrid.vue should match all specifications bellow', () => {
    const props = {
      tasks: [
        {
          name: 'Test 01',
          pending: false,
        },
        {
          name: 'Test 02',
          pending: true,
        },
      ],
    };
    const wrapper = shallowMount(TaskGrid, {
      'propsData': props,
    });
    expect(wrapper.isVueInstance()).toBeTruthy();

    expect(wrapper.classes('task-grid')).toBe(true);

    expect(wrapper.contains('div[class="no-task"]')).toBe(false);

    expect(wrapper).toMatchSnapshot();
  });

  it('TaskGrid.vue should contain a method deleteTask', () => {
    const idx = 1;
    const props = {
      tasks: [
        {
          name: 'Test 01',
          pending: false,
        },
        {
          name: 'Test 02',
          pending: true,
        },
      ],
    };
    const wrapper = shallowMount(TaskGrid, {
      'propsData': props,
    });
    expect(wrapper.isVueInstance()).toBeTruthy();

    wrapper.vm.$emit('task-deleted', idx);
    const emitted = wrapper.emitted();

    expect(emitted['task-deleted']).toBeTruthy()
    expect(emitted['task-deleted'].length).toBe(1)
    expect(emitted['task-deleted'][0]).toEqual([idx])

    expect(wrapper).toMatchSnapshot();
  });
});