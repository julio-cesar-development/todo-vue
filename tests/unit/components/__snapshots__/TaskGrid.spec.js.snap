// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskGrid.vue TaskGrid.vue should contain a method deleteTask 1`] = `
<div class="task-grid">
  <task-stub task="[object Object]"></task-stub>
  <task-stub task="[object Object]"></task-stub>
</div>
`;

exports[`TaskGrid.vue TaskGrid.vue should match all specifications bellow 1`] = `
<div class="task-grid">
  <task-stub task="[object Object]"></task-stub>
  <task-stub task="[object Object]"></task-stub>
</div>
`;
